From f19be09d2c8fe53676ee6ebff0be3173a87dc578 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Tue, 26 May 2015 23:41:13 +0800
Subject: [PATCH 0249/5242] MLK-10973 ARM: imx6: correct the soc_id on imx6qp

commit  1f6438bf11f85c1cdc74e83ccf38bdc4fb6d4a56 from
https://source.codeaurora.org/external/imx/linux-imx.git

On i.MX6QP, the soc_id exported to the /sys/devices/soc0/soc_id
should be 'i.MX6QP'.

Signed-off-by: Bai Ping <b51503@freescale.com>
(cherry picked from commit 5b478870b0b936a6c89574e6fe62f5537e31ecf0)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/cpu.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/cpu.c b/arch/arm/mach-imx/cpu.c
index 32969f3..1702485 100644
--- a/arch/arm/mach-imx/cpu.c
+++ b/arch/arm/mach-imx/cpu.c
@@ -127,7 +127,10 @@ struct device * __init imx_soc_device_init(void)
 		soc_id = "i.MX6SX";
 		break;
 	case MXC_CPU_IMX6Q:
-		soc_id = "i.MX6Q";
+		if (imx_get_soc_revision() == IMX_CHIP_REVISION_2_0)
+			soc_id = "i.MX6QP";
+		else
+			soc_id = "i.MX6Q";
 		break;
 	case MXC_CPU_IMX6UL:
 		soc_id = "i.MX6UL";
-- 
1.7.9.5

